python/cudf/cudf/tests/test_dataframe.py

```diff
@@ -9177,3 +9177,24 @@ def test_dataframe_rename_duplicate_column():
         ValueError, match="Duplicate column names are not allowed"
     ):
         gdf.rename(columns={"a": "b"}, inplace=True)
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        np.random.RandomState(seed=10).randint(-50, 50, (25, 30)),
+        np.random.RandomState(seed=10).random_sample((4, 4)),
+        np.array([1.123, 2.343, 5.890, 0.0]),
+        {"a": [1.123, 2.343, np.nan, np.nan], "b": [None, 3, 9.08, None]},
+    ],
+)
+@pytest.mark.parametrize("periods", (-2, -1, 0, 1, 2))
```

Comment:
Let's be consistent in using `list` to specify parameters. And `1` shifts seems too trivial if `2` is tested. Let's also add tests that goes beyond the length of the data (similar to what you did in `diff`).

```suggestion
@pytest.mark.parametrize("periods", [-5, -2, 0, 2, 5])
```